### PR TITLE
Add development environment setup in the CONTIBUTE markdown and move the readme info 

### PR DESCRIPTION
**Pull request by Jens-Co**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,3 +43,15 @@ Once you have added the file you can add an sidebar link, All sidebar links + su
         - title: PageChild sub link
           url: "page/pagetitle/#headerLink"
 ```
+
+### GeyserWiki  development setup locally
```
Comment (onebeastchris) [inline README.md:47]:
```suggestion
### GeyserWiki development - Local deployment 
```
double space; mayyybe better title? not fully sure about it tbh

File: README.md [hunk 2]
```diff
@@ -43,3 +43,15 @@ Once you have added the file you can add an sidebar link, All sidebar links + su
...
+
+* Download Ruby and Take the Ruby+Devkit from this page; https://rubyinstaller.org/downloads/
```
Comment (onebeastchris) [inline README.md:49]:
```suggestion
* Download and install Ruby using the Ruby+Devkit from https://rubyinstaller.org/downloads/.
```

File: README.md [hunk 3]
```diff
@@ -43,3 +43,15 @@ Once you have added the file you can add an sidebar link, All sidebar links + su
...
+
+* Once installed you will have to Run 'ridk install', this will open the RubyInstaller console.
```
Comment (onebeastchris) [inline README.md:51]:
```suggestion
* Once installed, run 'ridk install' in console to open the RubyInstaller.
```

File: README.md [hunk 4]
```diff
@@ -43,3 +43,15 @@ Once you have added the file you can add an sidebar link, All sidebar links + su
...
+Recommend choosing option 3; MSYS and MINGW after that ruby has been installed.
```
Comment (onebeastchris) [inline README.md:52]:
```suggestion
For this usecase, it is recommended to choose the third option, "MSYS and MINGW". After that, the Ruby installation is completed.
```
i'd mention that it's best for this usecase

File: README.md [hunk 5]
```diff
@@ -43,3 +43,15 @@ Once you have added the file you can add an sidebar link, All sidebar links + su
...
+
+* Open the GeyserWiki folder in VScode, in terminal you will have to install the bundle first 'bundle install' command.
+installing for the first time can take a while.
+
+* Once installed you can now start the server locally with command 'bundle exec jekyll serve'
```
Comment (onebeastchris) [inline README.md:57]:
```suggestion
* Once installed, you can start the webserver locally with the 'bundle exec jekyll serve' command in the VSC terminal. 
```

**Review by onebeastchris — CHANGES_REQUESTED** [review]
added some suggestions about how i'd phrase it. Also, might be worth removing layout/permalink from the "example page layout section" since that'd outdated

**Comment by Chew** [comment]
This should go in a CONTRIBUTING.md file tbh. No need for it to be in the README.

**Comment by onebeastchris** [comment]
@Chew if this should go into contributing.md, then the example should also be moved - imo, as this repo is only used to contribute, having a "contributing" section with all necessary details in the readme is enough.
But okay, if we move it, then we should move all of it - although that would leave a readme with basically just a description

**Comment by Chew** [comment]
> if this should go into contributing.md, then the example should also be moved

Correct.

> imo, as this repo is only used to contribute, having a "contributing" section with all necessary details in the readme is enough.

If that's the case then Geyser's main repo README should have contributing stuff there too, right? The point of a README is to tell people passing by what the point of this repo is. Sure it's just a wiki repo. We could do a Contributing section that says "If you wanna contribute see our guidelines." This is just normal practice in repos tho, especially since GitHub handles CONTRIBUTING.md files better, by telling people who are making PRs about it and the etc.

> But okay, if we move it, then we should move all of it - although that would leave a readme with basically just a description

So be it, READMEs don't need to be overly descriptive.

**Comment by Tim203** [comment]
Is it an idea to provide a docker-compose file instead? I don't think everyone has Ruby e.g. but it's quite more likely that they have Docker installed.

**Comment by onebeastchris** [comment]
personally, I found ruby simpler - I don't have docker set up atm
Maybe both would be an idea?

**Comment by rtm516** [comment]
> Is it an idea to provide a docker-compose file instead? I don't think everyone has Ruby e.g. but it's quite more likely that they have Docker installed.

On linux yes, docker would be better. But windows the ruby installer would be better.

I think this should be in `CONTRIBUTING.md` and split into setup for windows (ruby installer), linux (apt), docker and then having the shared things like `bundle install` in a section at the bottom.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Most documentation is in the docs folder where you can add new page's or edit the current ones.
+If a new page is being added make sure you add its page properties at the beginning of the file.
+Make sure that the file extension is .md in this case pagetitle.md
+
+### Example page layout
+
+```md
+---
+title: Page title
+page_sidebar: false "optional"
```
Comment (onebeastchris) [inline CONTRIBUTING.md:16]: i would put optional in brackets here, but that might be just me

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# Contributing
+
+If you would like to contribute to the wiki, please [open a pull request](https://github.com/GeyserMC/GeyserWiki/pulls).
+
+### Creating a new page
+
+Most documentation is in the docs folder where you can add new page's or edit the current ones.
```
Comment (onebeastchris) [inline CONTRIBUTING.md:7]:
```suggestion
Most documentation is in the docs folder where you can add new pages or edit the current ones.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+If a new page is being added make sure you add its page properties at the beginning of the file.
+Make sure that the file extension is .md in this case pagetitle.md
```
Comment (onebeastchris) [inline CONTRIBUTING.md:9]:
```suggestion
Make sure that the file extension is .md - in this case `pagetitle.md`.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+page_sidebar: false "optional"
+---
+
+# Page title
+This is an example page
+```
+
+### Adding page to sidebar 
+
+Once you have added the file you can add an sidebar link, All sidebar links + sub links are in the _data/toc.yml file
```
Comment (onebeastchris) [inline CONTRIBUTING.md:25]:
```suggestion
Once you have added the file you can add a sidebar link. All sidebar links + sub links are in the _data/toc.yml file.
```

**Review by onebeastchris — CHANGES_REQUESTED** [review]

**Review by onebeastchris — APPROVED** [review]

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+---
+title: Page title
+page_sidebar: false (optional)
+---
+
+# Page title
+This is an example page
+```
+
+### Adding page to sidebar
```
Comment (onebeastchris) [inline CONTRIBUTING.md:23]:
```suggestion
### Adding page to the sidebar 
```

**Review by onebeastchris — APPROVED** [review]
After these, LGTM

**Review by onebeastchris — APPROVED** [review]
thanks :)

**Pull request merged**